Classes/Controllers/PBGitIndexController.m

```diff
@@ -89,117 +89,31 @@ - (BOOL) allSelectedCanBeIgnored:(NSArray *)selectedFiles
 	return YES;
 }
 
-- (NSMenu *) menuForTable:(NSTableView *)table
-{
-	NSMenu *menu = [[NSMenu alloc] init];
-	NSArrayController *controller = table.tag == 0 ? unstagedFilesController : stagedFilesController;
-	NSArray *selectedFiles = controller.selectedObjects;
-	
-	NSUInteger numberOfSelectedFiles = selectedFiles.count;
-
-	if (numberOfSelectedFiles == 0)
-	{
-		return menu;
-	}
-	
-	// Stage/Unstage changes
-	if (table.tag == 0) {
-		NSString *stageTitle = numberOfSelectedFiles == 1
-			? [NSString stringWithFormat:NSLocalizedString( @"Stage “%@”", @"Stage single file contextual menu item" ), [self getNameOfFirstSelectedFile:selectedFiles]]
-			: [NSString stringWithFormat:NSLocalizedString( @"Stage %i Files", @"Stage multiple files contextual menu item"), numberOfSelectedFiles ];
-		NSMenuItem *stageItem = [[NSMenuItem alloc] initWithTitle:stageTitle action:@selector(stageFilesAction:) keyEquivalent:@"s"];
-		stageItem.target = self;
-		[menu addItem:stageItem];
-	}
-	else if (table.tag == 1) {
-		NSString *stageTitle = numberOfSelectedFiles == 1
-			? [NSString stringWithFormat:NSLocalizedString( @"Unstage “%@”", @"Unstage single file contextual menu item" ), [self getNameOfFirstSelectedFile:selectedFiles]]
-			: [NSString stringWithFormat:NSLocalizedString( @"Unstage %i Files", @"Unstage multiple files contextual menu item"), numberOfSelectedFiles ];
-		NSMenuItem *unstageItem = [[NSMenuItem alloc] initWithTitle:stageTitle action:@selector(unstageFilesAction:) keyEquivalent:@"u"];
-		unstageItem.target = self;
-		[menu addItem:unstageItem];
-	}
-
-	NSString *openTitle = numberOfSelectedFiles == 1
-		? [NSString stringWithFormat:NSLocalizedString( @"Open ”%@“", @"Open single file contextual menu item" ), [self getNameOfFirstSelectedFile:selectedFiles]]
-		: [NSString stringWithFormat:NSLocalizedString( @"Open %i Files", @"Open multiple files contextual menu item"), numberOfSelectedFiles ];
-	NSMenuItem *openItem = [[NSMenuItem alloc] initWithTitle:openTitle action:@selector(openFilesAction:) keyEquivalent:@""];
-	openItem.target = self;
-	[menu addItem:openItem];
-
-	// Attempt to ignore
-	if ([self allSelectedCanBeIgnored:selectedFiles]) {
-		NSString *ignoreText = numberOfSelectedFiles == 1
-			? [NSString stringWithFormat:NSLocalizedString( @"Ignore ”%@“", @"Ignore single file contextual menu item" ), [self getNameOfFirstSelectedFile:selectedFiles]]
-			: [NSString stringWithFormat:NSLocalizedString( @"Ignore %i Files", @"Ignore multiple files contextual menu item"), numberOfSelectedFiles ];
-		NSMenuItem *ignoreItem = [[NSMenuItem alloc] initWithTitle:ignoreText action:@selector(ignoreFilesAction:) keyEquivalent:@""];
-		ignoreItem.target = self;
-		[menu addItem:ignoreItem];
-	}
++ (NSString *) getNameOfFirstFile:(NSArray<PBChangedFile *> *) selectedFiles {
```

Comment:
Can you move those class methods in `PBTableViewWhatWasItsName` instead of keeping them here ?